### PR TITLE
ADEN-2262 Add waiter

### PR DESCRIPTION
**Pull request by drets**

### CODE REVIEW

**Comment by nandy-andy** [comment]
LGTM :+1: But as we discussed with @jacekjursza we can start thinking if there are other and better solutions than waiting (nothing comes to my mind ATM but I wanted to seed the idea).

**Comment by drets** [comment]
@nandy what's wrong with waiting till page is loaded?

**Comment by nandy-andy** [comment]
Nothing really :smile_cat: We were just talking that if we put everywhere `waitFor*` it could make the execution time for our tests bigger. But in this specific case I can't see other solution, so you did a good job, sir :bow: It's more philosophical topic we had today with JJ :dancer:

Just focus on the convention and once you're back here we can talk more in the kitchen about our tests' execution time :dash:

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsHopObject.java [hunk 1]
```diff
@@ -22,6 +23,7 @@
 
   private static final String POST_MESSAGE_SCRIPT_XPATH =
       "//script[contains(text(), 'parent.postMessage')]";
+  private static final int AD_SUCCESS_TIMEOUT_SEC = 15;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsHopObject.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsHopObject.java [hunk 2]
```diff
@@ -20,9 +21,9 @@
  */
 public class AdsHopObject extends AdsBaseObject {
 
-  private static final String POST_MESSAGE_SCRIPT_XPATH =
+  private final static String POST_MESSAGE_SCRIPT_XPATH =
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsHopObject.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsHopObject.java [hunk 3]
```diff
@@ -33,13 +34,21 @@ public AdsHopObject(WebDriver driver, String page) {
...
 
-  public AdsHopObject verifyClassHidden(String slotName, String src) {
-    WebElement testedDiv = getTestedDiv(slotName, src);
-    Assertion.assertEquals(testedDiv.getAttribute("class").trim(), "hidden");
-    return this;
+  public void verifyClassHidden(final String slotName, final String src) {
+    new WebDriverWait(driver, AD_SUCCESS_TIMEOUT_SEC).until(new ExpectedCondition<Boolean>() {
+      @Override
+      public Boolean apply(WebDriver driver) {
+        return getTestedDiv(slotName, src).getAttribute("class").trim().equals("hidden");
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsHopObject.java]: MAJOR Move the "hidden" string literal on the left side of this string comparison. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsHopObject.java [hunk 4]
```diff
@@ -20,9 +21,9 @@
  */
 public class AdsHopObject extends AdsBaseObject {
 
-  private static final String POST_MESSAGE_SCRIPT_XPATH =
+  private final static String POST_MESSAGE_SCRIPT_XPATH =
       "//script[contains(text(), 'parent.postMessage')]";
-
+  private final static int AD_SUCCESS_TIMEOUT_SEC = 15;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsHopObject.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsHopObject.java [hunk 5]
```diff
@@ -22,8 +23,8 @@
 
   private static final String POST_MESSAGE_SCRIPT_XPATH =
       "//script[contains(text(), 'parent.postMessage')]";
-
-  private final static ImmutableMap<String, String> dfpSrc =
+  private static final int AD_SUCCESS_TIMEOUT_SEC = 15;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsHopObject.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsHopMercury.java [hunk 6]
```diff
@@ -11,18 +11,18 @@
  * @author drets
  * @ownership AdEng
  */
-public class TestAdsHop extends TemplateNoFirstLoad {
+public class TestAdsHopMercury extends TemplateNoFirstLoad {
```
Comment (nandy-andy) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsHopMercury.java]: Isn't this test more generic and can be executed on Mercury and WikiaMobile?

Comment (nandy-andy) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsHopMercury.java]: OK, it runs now on Mercury, so this name makes sense.

**Comment by nandy-andy** [comment]
Hello Sonar! :) The changes LGTM :+1:

**Pull request merged**